### ISSUE TITLE
Saving an `Optimizer` initially instantiated with a `float` learning_rate and later replaced with a `LearningRateSchedule` includes an extra Variable.

### ISSUE THREAD

**Issue by briango28** [keras-team/keras#19492]
This is technically a bug but niche enough that most people won't ever be effected.

When instantiating a keras [`Optimizer`](https://keras.io/api/optimizers/), the `__init__()` method of the base class [registers its learning rate as a keras `Variable`](https://github.com/keras-team/keras/blob/v3.2.1/keras/optimizers/base_optimizer.py#L133) if the given `learning_rate` is of type `float`.
However, when [setting](https://github.com/keras-team/keras/blob/v3.2.1/keras/optimizers/base_optimizer.py#L503) the `learning_rate` to something that isn't a primitive `float`, there is no mechanism to remove the `Variable` that was created.

This results in an extra unused variable written to disk when saving a model that uses this optimizer, while the exported configuration dict does not make note of this fact due to [`get_config()`](https://github.com/keras-team/keras/blob/v3.2.1/keras/optimizers/base_optimizer.py#L742) only checking the current type of `self._learning_rate`.

When loading a model saved in this way, the optimizer will fail to load variables from disk, [warning](https://github.com/keras-team/keras/blob/v3.2.1/keras/optimizers/base_optimizer.py#L548) that there is a mismatch between the number of variables in the instantiated object and the number of stored variables.

The following colab notebook illustrates this issue.
https://colab.research.google.com/drive/17HhZeIR-IShZlFywvIDNF9XBPyO71cj9?usp=sharing
Note the `UserWarning` raised in the last cell.
> UserWarning: Skipping variable loading for optimizer 'adam', because it has 5 variables whereas the saved optimizer has 6 variables.

Current workarounds for dynamically changing a learning rate from fixed to scheduled are
a) manually remove or exclude the learning rate `Variable` before saving.
b) use a [`LearningRateScheduler`](https://keras.io/api/callbacks/learning_rate_scheduler/) callback instead

**Comment by fchollet** [comment]
Thanks for the report. I've fixed it at HEAD.
